src/pointer-analysis/dynamic_object_name.cpp

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************\
+
+Module: Dynamic object name
+
+Author: Marius-Constantin Melemciuc
+
+Date: April 2017
+
+@ Copyright Diffblue, Ltd.
+
+\*******************************************************************/
+
+#include "dynamic_object_name.h"
+
+/*******************************************************************\
+
+Function: get_dynamic_object_prefix
+
+  Inputs:
+
+ Outputs: The prefix of the dynamic-object,
+          composed of "value_set::dynamic_object".
+
+ Purpose: To generate a prefix for dynamic-objects suitable for use
+          in the LHS of value-set maps.
+
+\*******************************************************************/
+
+const std::string &get_dynamic_object_prefix()
+{
+  static std::string prefix="value_set::dynamic_object";
```

Comment:
The return value of this function is normally used as a `std::string`, not a `const char *` (e.g. added to other string constants) so returning a reference to a `static std::string` avoids constructing one at each call site. Alternatives would be to construct a `std::string` in this function from a string constant or to return a `const char *` and construct a `std::string` at the call site.